topbeat/system/process.go

```diff
@@ -211,20 +235,22 @@ func (procStats *ProcStats) GetProcStats() ([]common.MapStr, error) {
 			cmdline = previousProc.CmdLine
 		}
 
-		process, err := GetProcess(pid, cmdline)
+		process, err := newProcess(pid)
 		if err != nil {
 			logp.Debug("topbeat", "Skip process pid=%d: %v", pid, err)
 			continue
 		}
 
 		if procStats.MatchProcess(process.Name) {
+			err = process.getDetails(cmdline)
+			if err != nil {
+				logp.Err("Error getting process details. pid=%d: %v", process.Pid, err)
+				continue
+			}
 
 			newProcs[process.Pid] = process
 
-			last, ok := procStats.ProcsMap[process.Pid]
-			if ok {
-				procStats.ProcsMap[process.Pid] = process
-			}
```

Comment:
@tsg I think you can simply remove the entire `if` as ProcsMap is updated in "procStats.ProcMap = newProcs" below.